imap_processing/tests/spice/test_pointing_frame_handler.py

```diff
@@ -0,0 +1,177 @@
+    """Tests get_et_times function."""
+    kernels, ck_kernel = create_kernel_list
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_et_times(str(ck_kernel[0]))
+
+    assert et_start == 802008069.184905
+    assert et_end == 802094467.184905
+    assert len(et_times) == 57599
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_average_quaternions(et_times, create_kernel_list):
+    """Tests average_quaternions function."""
+
+    kernels, ck_kernel = create_kernel_list
+    with spice.KernelPool(kernels):
+        q_avg, z_eclip_time = average_quaternions(et_times)
+
+    # Generated from MATLAB code results
+    q_avg_expected = np.array([-0.6838, 0.5480, -0.4469, -0.1802])
+    np.testing.assert_allclose(q_avg, q_avg_expected, atol=1e-4)
+    assert len(z_eclip_time) == 57599
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_create_rotation_matrix(et_times, kernel_path):
+    """Tests create_rotation_matrix function."""
+
+    kernels = [str(file) for file in kernel_path.iterdir()]
+
+    with spice.KernelPool(kernels):
+        rotation_matrix, z_avg = create_rotation_matrix(et_times)
+
+    rotation_matrix_expected = np.array(
+        [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
+    )
+    z_avg_expected = np.array([0.4136, 0.9104, 0.0000])
+
+    np.testing.assert_allclose(z_avg, z_avg_expected, atol=1e-4)
+    np.testing.assert_allclose(rotation_matrix, rotation_matrix_expected, atol=1e-4)
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_create_pointing_frame(monkeypatch, kernel_path, create_kernel_list):
+    """Tests create_pointing_frame function."""
+    monkeypatch.setenv("EFS_MOUNT_PATH", str(kernel_path))
+    _, ck_kernel = create_kernel_list
+    create_pointing_frame()
+
+    # After imap_dps.bc has been created.
+    kernels = [str(file) for file in kernel_path.iterdir()]
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_et_times(str(ck_kernel[0]))
+
+        rotation_matrix_1 = spice.pxform("ECLIPJ2000", "IMAP_DPS", et_start + 100)
+        rotation_matrix_2 = spice.pxform("ECLIPJ2000", "IMAP_DPS", et_start + 1000)
+
+    # All the rotation matrices should be the same.
+    assert np.array_equal(rotation_matrix_1, rotation_matrix_2)
+
+    # Nick Dutton's MATLAB code result
+    rotation_matrix_expected = np.array(
+        [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
+    )
+    np.testing.assert_allclose(rotation_matrix_1, rotation_matrix_expected, atol=1e-4)
+
+    # Verify imap_dps.bc has been created.
+    assert (kernel_path / "imap_dps.bc").exists()
+
+
+@pytest.mark.skip(reason="Plotting only")
+def test_declination_plot(create_kernel_list, et_times):
+    """Tests Inertial z axis and provides visualization."""
```

Comment:
I don't think think this is "testing", it is a helper plot function, but there are no image comparisons or assertions going on.